Library/ViewModels/CommentRepliesViewModel.swift

```diff
@@ -94,18 +97,35 @@ public final class CommentRepliesViewModel: CommentRepliesViewModelType,
         return (url, canPostComment, false, inputAreaBecomeFirstResponder)
       }
 
-    let repliesEvent = rootComment
-      .switchMap { comment in
-        AppEnvironment.current.apiService.fetchCommentReplies(
-          query: commentRepliesQuery(withCommentId: comment.id)
-        )
-        .materialize()
-      }
+    let totalCountProperty = MutableProperty<Int>(0)
+
+    // TODO: Handle isLoading from here
+    let (replies, _, _, _) = paginate(
+      requestFirstPageWith: rootComment,
+      requestNextPageWhen: self.viewMoreRepliesCellWasTappedProperty.signal,
+      clearOnNewRequest: true,
+      valuesFromEnvelope: { [totalCountProperty] envelope -> [Comment] in
+        totalCountProperty.value = envelope.totalCount
+        return envelope.replies
+      },
+      cursorFromEnvelope: { envelope in
+        (envelope.comment, envelope.cursor)
+      },
+      requestFromParams: { comment in
+        AppEnvironment.current.apiService
+          .fetchCommentReplies(query: commentRepliesQuery(withCommentId: comment.id))
+      },
+      requestFromCursor: {
+        AppEnvironment.current.apiService
+          .fetchCommentReplies(query: commentRepliesQuery(withCommentId: $0.0.id, before: $0.1))
+      },
+      // only return new pages, we'll concat them ourselves
+      concater: { _, value in value }
```

Comment:
So I guess `totalCountProperty.value = value` wouldn't work here?

Reply:
We decided to set it in `valuesFromEnvelope` because it only needs to be set once and we don't want the signal to emit any more times than necessary (the value of totalCount never changes).